kythe/java/com/google/devtools/kythe/extractors/java/bazel/JavaExtractor.java

```diff
@@ -49,17 +50,30 @@
 
 /** Java CompilationUnit extractor using Bazel's extra_action feature. */
 public class JavaExtractor {
+
   private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String KZIP_OUTPUT_FORMAT = "kzip";
+  private static final String KINDEX_OUTPUT_FORMAT = "kindex";
 
   public static void main(String[] args) throws IOException, ExtractionException {
-    if (args.length != 3) {
-      System.err.println("Usage: java_extractor extra-action-file output-file vname-config");
+    JsonUtil.usingTypeRegistry(JavaCompilationUnitExtractor.JSON_TYPE_REGISTRY);
+
+    if (args.length != 4) {
+      System.err.println(
+          "Usage: java_extractor extra-action-file output-file vname-config output-format");
```

Comment:
Why not infer `output-format` from `output-file`'s extension?

Reply:
same as below